core/arch/arm/kernel/generic_boot.c

```diff
@@ -726,6 +740,8 @@ static void init_fdt(unsigned long phys_fdt)
 		     phys_fdt, ret);
 		panic();
 	}
+
+	dt_blob_addr = fdt;
```

Comment:
This pointer isn't valid forever. It should be cleared when OP-TEE has initialized and exits the first time.

Reply:
I need to parse the dtb to get fsl,ext-reset-output in wdog node. In this patchset, I use driver_init(xxx_wdog_xx) to handle that. If not expose dtb here, I am not able to parse dtb for the drivers. Could we expose fdt in init_fdt, and destroy the pointer when first back to normal world? Or we create a copy of the dtb in optee secure side?

Reply:
If you clear it before `init_primary_helper()` (after calling `init_teecore()`) it should be OK.

Reply:
Thanks. I'll clear it as you suggested.